assembly/src/ast/mod.rs

```diff
@@ -145,10 +162,15 @@ impl ProgramAst {
     pub fn parse(source: &str) -> Result<ProgramAst, ParsingError> {
         let mut tokens = TokenStream::new(source)?;
         let imports = parse_imports(&mut tokens)?;
+        if imports.len() > MAX_IMPORTS {
+            return Err(ParsingError::too_many_imports(imports.len(), MAX_IMPORTS));
+        }
```

Comment:
This check should probably happen inside `parse_imports()`.

Reply:
It already does... I'll remove it here.